crates/miden-tx/tests/integration/scripts/compatibility.rs

```diff
@@ -0,0 +1,490 @@
+    // --------------------------------------------------------------------------------------------
+
+    let mut mock_chain = MockChain::new();
+    let faucet_account = mock_chain.add_existing_faucet(Auth::BasicAuth, "POL", 200u64, None);
+    let faucet_account_interface = AccountInterface::from(faucet_account.account());
+
+    let sender_account_id =
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN_2.try_into().unwrap();
+    let serial_num =
+        RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]).draw_word();
+    let tag = NoteTag::from_account_id(faucet_account.id(), NoteExecutionMode::Local).unwrap();
+    let metadata = NoteMetadata::new(
+        sender_account_id,
+        NoteType::Public,
+        tag,
+        NoteExecutionHint::always(),
+        Default::default(),
+    )
+    .unwrap();
+    let vault = NoteAssets::new(vec![FungibleAsset::mock(100)]).unwrap();
+
+    let compatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.miden::contracts::faucets::basic_fungible->fungible_faucet
+
+        begin
+            push.1
+            if.true 
+                call.fungible_faucet::distribute
+                call.fungible_faucet::burn
+            else
+                call.wallet::receive_asset
+                call.wallet::create_note
+                call.wallet::move_asset_to_note
+            end
+        end
+    ";
+    let note_script =
+        NoteScript::compile(compatible_source_code, TransactionKernel::testing_assembler())
+            .unwrap();
+    let recipient = NoteRecipient::new(serial_num, note_script, NoteInputs::default());
+    let compatible_custom_note = Note::new(vault.clone(), metadata, recipient);
+    assert_eq!(
+        CheckResult::Maybe,
+        faucet_account_interface.can_consume(&compatible_custom_note)
+    );
+
+    let incompatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.miden::contracts::faucets::basic_fungible->fungible_faucet
+
+        begin
+            push.1
+            if.true 
+                call.fungible_faucet::distribute
+                call.wallet::receive_asset
+            else
+                call.fungible_faucet::burn
+                call.wallet::create_note
+                call.wallet::move_asset_to_note
+            end
+        end
+    ";
+    let note_script =
+        NoteScript::compile(incompatible_source_code, TransactionKernel::testing_assembler())
+            .unwrap();
+    let recipient = NoteRecipient::new(serial_num, note_script, NoteInputs::default());
+    let incompatible_custom_note = Note::new(vault, metadata, recipient);
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&incompatible_custom_note));
+
+    // STATEFUL CHECK
+    // --------------------------------------------------------------------------------------------
+
+    // TODO: implement
+}
+
+#[test]
+fn test_custom_account_custom_notes() {
```

Comment:
Can you add 1-2 sentence comments explaining the purpose of the test, what result it expects and why?